posts/2022-07-12-components-with-dynamic-attributes.md

```diff
@@ -0,0 +1,272 @@
+
+Now that we want the cards in each section to have different colors, how can we make it possible keeping in mind that we are using reusable components.
+
+
+Let’s begin:
+
+## 1.Create Board LiveView
+
+```elixir
+defmodule SampleWeb.BoardLive do
+  use  SampleWeb, :live_view
+
+  def mount(_params, _session, socket) do
+    work_cards = [
+      %{task: "deploy to production", id: "#{1}-work"},
+      %{task: "code challenge", id: "#{2}-work"},
+      %{task: "plan community events", id: "#{3}-work"}
+    ]    
+
+    house_cards = [
+      %{task: "wash my dog", id: "#{1}-house"},
+      %{task: "sweep the house", id: "#{2}-house"},
+      %{task: "tidy my bedroom", id: "#{3}-house"}
+    ]    
+
+    school_cards = [
+      %{task: "group discussion", id: "#{1}-school"},
+      %{task: "submit assignment", id: "#{2}-school"},
+      %{task: "work on school project", id: "#{3}-school"}
+    ]
+
+    {:ok,
+     assign(socket, work_cards: work_cards, house_cards: house_cards, school_cards: school_cards)}
+
+  end
+
+  def render(assigns) do
+    ~H"""
+    <h1>Board</h1>
+    <h2>Weekly Board Task </h2>
+    <div class="row">
+      <div class="column">
+        <h3>Work</h3>
+        <%= for card <- @work_cards do %> 
+          <.card card={card} />
+        <% end %>
+       </div>
+       <div class="column">
+         <h3>House</h3>
+         <%= for card <- @house_cards do %> 
+           <.card card={card} />
+         <% end %>
+       </div>
+       <div class="column">
+         <h3>School</h3>
+         <%= for card <- @school_cards do %> 
+           <.card card={card} />
+         <% end %>
+        </div>
+      </div>
+    """
+end
+
+def card(assigns) do
+    ~H"""
+    <div>
+      <div class="column">
+        <%= @card.task %>
+      </div>
+    </div>
+    """
+  end
+end
+
+```
+
+I have created a BoardLive page that renders each card component for house, work and school section.
+
+The card components have been defined using [function components](https://hexdocs.pm/phoenix_live_view/0.16.0/Phoenix.Component.html) mechanism, which is useful for reusing markup in our LiveView.
+
+**NB:** I’m not getting into the details of how the user should add their weekly task. In this example I have hard-coded the task assuming that the user had already added their task. We are actually displaying the task added to the cards .
+
+Our weekly planning task board should look something similar to this when we open our browser:
+
+
+![board plan output](/images/board-plan-output.png)
+
+
+## 2.Problem
+
+  1. We want the cards in each section to have a different color from one another . For example, cards in work section can be blue, in house can be green and in school section can be yellow.
+
+  2. Card component defined earlier should also be maintained.
```

Comment:
I'd re-phrase this as: `We should be able to use the same `card/1` function component we defined earlier, while still ensuring that cards can be a different color in the different columns.`